internal/cmd/provider/create_version.go

```diff
@@ -0,0 +1,165 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+	"github.com/urfave/cli/v2"
+
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+)
+
+func createVersion() cli.ActionFunc {
+	return func(cliCtx *cli.Context) error {
+		// Assuming that spacectl is ran from the root of the repository,
+		// containing the release artifacts in the "dist" directory.
+		dir := "dist"
+
+		if cliCtx.IsSet(flagGoReleaserDir.Name) {
+			dir = cliCtx.String(flagGoReleaserDir.Name)
+		}
+
+		providerType := cliCtx.String(flagProviderType.Name)
+
+		fmt.Println("Retrieving release data from ", dir)
+		versionData, err := BuildGoReleaserVersionData(dir)
+		if err != nil {
+			return errors.Wrap(err, "invalid release data")
+		}
+
+		checksumsFile, err := versionData.Artifacts.ChecksumsFile()
+		if err != nil {
+			return err
+		}
+
+		checksumsFileChecksum, err := checksumsFile.Checksum(dir)
+		if err != nil {
+			return errors.Wrap(err, "could not calculate checksum of checksums file")
+		}
+
+		signatureFile, err := versionData.Artifacts.SignatureFile()
+		if err != nil {
+			return err
+		}
+
+		signatureFileChecksum, err := signatureFile.Checksum(dir)
+		if err != nil {
+			return errors.Wrap(err, "could not calculate checksum of signature file")
+		}
+
+		var createMutation struct {
+			CreateTerraformProviderVersion struct {
+				SHA256SumsUploadURL    string `graphql:"sha256SumsUploadURL"`
+				SHA256SumsSigUploadURL string `graphql:"sha256SumsSigUploadURL"`
+				Version                struct {
+					ID string `graphql:"id"`
+				} `graphql:"version"`
+			} `graphql:"terraformProviderVersionCreate(provider: $provider, input: $input)"`
+		}
+
+		protocolVersions := []string{"5.0"}
```

Comment:
Should this maybe be a default value for the flag? This way, it would be automatically visible in usage print.

Reply:
Does this library even support default values?

Reply:
Oh, true. The v2.16.3 that we're using seems not to support them. I was looking at the cli repo, and it seems that v2.23.5 already does.

Reply:
OK I can try to bump the library, using default values will obviously be much cleaner.